components/api/src/test/java/com/hotels/styx/api/HttpResponseTest.java

```diff
@@ -310,6 +314,30 @@ public void rejectsInvalidContentLength() {
                 .build();
     }
 
+    @Test
+    public void encodesBodyWithCharset() throws InterruptedException, ExecutionException, TimeoutException {
+        StyxObservable<String> o = StyxObservable.of("Hello, World!");
+
+        byte[] bytesUtf8 = response()
+                .body(o, UTF_8)
+                .build()
+                .toFullResponse(1_000_000)
+                .map(FullHttpResponse::body)
+                .asCompletableFuture()
+                .get(1, SECONDS);
+
+        byte[] bytesUtf16 = response()
+                .body(o, UTF_16)
+                .build()
+                .toFullResponse(1_000_000)
+                .map(FullHttpResponse::body)
+                .asCompletableFuture()
+                .get(1, SECONDS);
+
+        assertThat(bytesUtf8, is("Hello, World!".getBytes(UTF_8)));
+        assertThat(bytesUtf16, is("Hello, World!".getBytes(UTF_16)));
```

Comment:
A minor comment:
The test case intention would be clearer if the response object appeared in the assertion like so:

    `assertThat(utf8Response.body(), is("Hello, World!".getBytes(UTF_8)))`;